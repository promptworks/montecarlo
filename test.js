const fs = require('fs');

/**
 * Stub out Math.random, because it's hard to test.
 */
global.Math.random = (() => {
  let i = 0;
  return () => ((i++) % 10) / 10;
})();

/**
 * Who needs modules anyway?
 */
eval(fs.readFileSync('./index.js').toString());

const XS = [.25, .5];
const S  = [.5, 1];
const M  = [1, 3];
const L  = [3, 5];
const XL = [5, 10];

const dataset = [
  M, L, M, M, L, XL, M, XL, L, M,
  M, L, M, M, L, S, M, M
];

test('generates a distribution', () => {
  expect(montecarlo(dataset, 25, 10)).toMatchSnapshot();
});

test('generates the correct number of results', () => {
  expect(montecarlo(dataset, 25, 10)).toHaveLength(25);
});
