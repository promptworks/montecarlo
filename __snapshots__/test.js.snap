// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates a distribution 1`] = `
Array [
  52.95,
  52.85,
  52.75,
  54.15,
  57.05,
  52.95,
  52.85,
  52.75,
  54.15,
  57.05,
  52.95,
  52.85,
  52.75,
  54.15,
  57.05,
  52.95,
  52.85,
  52.75,
  54.15,
  57.05,
  52.95,
  52.85,
  52.75,
  54.15,
  57.05,
]
`;
